src/screens/KeyBasedAssetTransfer/KeyBasedAssetTransferEditAmount.js

```diff
@@ -17,236 +17,114 @@
-};
+// Utils
+import { mapAssetDataToAssetOption } from 'utils/assets';
+import { BigNumber, formatAmount } from 'utils/common';
+import { spacing } from 'utils/variables';
 
-const ErrorHolder = styled.View`
-  width: 100%;
-  padding: 0 ${spacing.mediumLarge}px;
-`;
+// types
+import type { AssetData } from 'models/Asset';
 
-const ErrorText = styled(BaseText)`
-  color: ${themedColors.negative};
-  ${fontStyles.small};
-  width: 100%;
-  text-align: right;
-`;
+function KeyBasedAssetTransferEditAmount() {
+  const navigation = useNavigation();
+  const assetData: ?AssetData = navigation.getParam('assetData');
+  const initialValue: ?number = navigation.getParam('value');
 
-const AmountInputWrapper = styled.View`
-  height: 70px;
-  justify-content: center;
-  min-width: 180px;
-`;
+  const inputRef = React.useRef();
 
-const AmountInput = styled(TextInput)`
-  ${fontSizes.small};
-  text-align: right;
-  max-width: 200px;
-`;
+  const [value, setValue] = React.useState(initialValue != null ? formatAmount(initialValue, assetData?.decimals) : '');
+  const [isValid, setIsValid] = React.useState(true);
 
-const KeyBasedAssetTransferEditAmount = ({
-  keyBasedAssetsToTransfer,
-  navigation,
-  isFetchingAvailableBalances,
-  availableBalances,
-  fetchAvailableBalancesToTransfer,
-  removeKeyBasedAssetToTransfer,
-  addKeyBasedAssetToTransfer,
-}: Props) => {
-  const [inSearchMode, setInSearchMode] = useState(false);
-  const [updatedValues, setUpdatedValues] = useState({});
-  const [errorMessages, setErrorMessages] = useState({});
-  const [searchQuery, setSearchQuery] = useState('');
-
-  const onAvailableBalancesRefresh = () => {
-    if (isFetchingAvailableBalances) return;
-    fetchAvailableBalancesToTransfer();
-  };
+  const dispatch = useDispatch();
+  const keyWalletBalances = useRootSelector((root) => root.keyBasedAssetTransfer.availableBalances);
+  const fiatCurrency = useFiatCurrency();
+  const rates = useRates();
 
-  const handleAmountChange = (text: string, assetData: AssetData) => {
-    const { token: symbol, decimals } = assetData;
-    const balance = getBalance(availableBalances, symbol);
-    const amountFormatted = text.toString().replace(/,/g, '.');
-    const amount = parseNumber(text);
-
-    let errorMessage;
-    if (!isValidNumber(text.toString())) {
-      errorMessage = t('error.amount.invalidNumber');
-    } else if (amount > balance) {
-      errorMessage = t('error.amount.exceedBalance');
-    } else if (!isValidNumberDecimals(amount, decimals)) {
-      errorMessage = t('error.amount.shouldNotHaveDecimals');
-    }
-
-    // resets or sets new
-    setErrorMessages({ ...errorMessages, [symbol]: errorMessage });
-
-    const updated = { assetData, amount: amountFormatted || 0 };
-    setUpdatedValues({ ...updatedValues, [symbol]: updated });
-  };
+  React.useEffect(() => {
+    InteractionManager.runAfterInteractions(() => {
+      inputRef.current?.focus();
+    });
```

Comment:
IMO, in this case using `useEffect` `{}` helps format the code so that it's easier to read, as for `useEffect` the dependency array (second argument) is quite important and should be clearly visible.

Reply:
Removed the inner `{}` but kept the `useEffect` `{}` as for it's important for deps array (second arg to `useEffect` should be clearly visible).

Reply:
Totally agree. That dependancy array has a lot of behaviour changes hidden under it!